.circleci/config.yml

```
@@ -141,7 +141,7 @@ jobs:
           name: install docker
           command: apt-get update -q && apt-get install -q -y docker.io
       - setup_remote_docker:
-          version: 19.03.13
+          version: 20.10.21
```

Could you try this?:
```suggestion
          version: 20.10.18
```

According to https://circleci.com/docs/building-docker-images/#docker-version, that appears to be the latest version available.

Note that I just cancelled the [` build-docker-image` ](https://circleci.com/gh/emscripten-core/emsdk/9824) job on this PR, as it's stuck almost 17 hours now (probably because of this).

---

Thanks for the tip @kleisauke. I've now tried to update the PR accordingly. Please let me know if you also want me to decouple the Docker engine update from the Ubuntu upgrade by submitting it in a separate PR that will have to be merged _before_ this one.

---

Great! It seems now to fail due to `Unable to locate package libidn11`, I think replacing `libidn11` in the Dockerfile with `libidn12` would fix that.

> Please let me know if you also want me to decouple the Docker engine update from the Ubuntu upgrade by submitting it in a separate PR that will have to be merged _before_ this one.

I think it's probably fine to do this in the same PR.

---

... although, perhaps we can drop that dependency, I'm not sure why it was added in the first place (maybe it was needed by OpenJDK 11, which is needed by Closure Compiler?). Let me investigate that as a possible follow-up.